common/models/user.js

```diff
@@ -301,6 +301,25 @@ module.exports = function(User) {
     return fn.promise;
   };
 
+  User.observe('before delete', function(ctx, next) {
+    var AccessToken = ctx.Model.app.models.AccessToken;
```

Comment:
Oh, hang on, that won't work. `accessTokens` is a property of model instance, which is not available when running a bulk delete via `User.destroyAll`.

You need to get the AccessToken model via `ctx.Model.relations.accessTokens.modelTo`.

Now I think we should support the general use case, where
 - the User model can have primary key called differently than `id`, possibly using a primary key composed of several properties, 
 - `User.destroyAll` can be called with arbitrary filter, e.g. `where={isPreferred:false, country: 'CZ' }`

In such case, it will be very difficult to build an `AccessToken` query that would match all tokens belonging to (and only to) the users matched by `ctx.where`.

Therefore I am proposing to use a different approach, similar to what you did:
 - first, find all deleted users
 - then delete all access tokens of these users

There is a catch though: while we are deleting the access tokens, it may happen that more records matching the original `where` clause are added by other clients. As a result, we would delete these new users, but not their access token.

The solution is to rewrite `ctx.where` to use `inq` filter with the list of the ids of the users we found.

```js
var AccessToken = ctx.Model.relations.accessTokens.modelTo;
ctx.Model.find({ where: ctx.where }, function(err, list) {
  if (err) return next(err);

  var pkName = ctx.Model.definition.idName() || 'id';
  var ids = list.map(function(u) { return u[pkName]; });
  ctx.where = {};
  ctx.where[pkName] = { inq: ids };

  AccessToken.destroyAll({ userId: { inq: ids } }, next);
});
```

(The code may not work OOTB.)

Also please add a unit-test to verify that access tokens are deleted when users are removed via `User.deleteAll`.